### PR TITLE
Switch to the big-int library we use in go-state-types

### LINKED ISSUE #517: Correctly handle sign in big-int serialization
cbor-gen ignores the sign when encoding `big.Int` from go stdlib. Everywhere else in Filecoin uses the one from [go state types](https://github.com/filecoin-project/go-state-types/blob/9c9af5cce8781845379e7abfa9c32bd06f9769a1/big/int.go#L15-L17) which encodes it as a leading 0/1 byte.

So, the question is what to do for the normal power table? We don't actually care there, but I assume we want to be consistent?

### PR DESCRIPTION
**Pull request by Stebalien**
Previously, we dropped the sign. Unfortunately, the only ways to do this were to:

1. Replace the StoragePower type (what I did here).
2. Hack marshaling/unmarshaling for anything containing this type (which I really didn't want to do).

It has some warts, but it works well in practice and works around some issues regarding nil encoding/decoding.

fixes #517
replaces #518

### CODE REVIEW

File: big/int.go [hunk 1]
```diff
@@ -0,0 +1,350 @@
+package big
```
Comment (Stebalien) [inline big/int.go:1]: Entirely copied from go-state-types. We can discuss depending on that library later, fixing cbor-gen, etc.

File: gpbft/powertable.go [hunk 2]
```diff
@@ -59,7 +60,7 @@ func (p PowerEntries) Len() int {
 // This ordering is guaranteed to be stable, since a valid PowerTable cannot contain entries with duplicate IDs; see Validate.
 func (p PowerEntries) Less(i, j int) bool {
 	one, other := p[i], p[j]
-	switch cmp := one.Power.Cmp(other.Power); {
+	switch cmp := big.Cmp(one.Power, other.Power); {
```
[the remark this answers is not in the document]
Comment (ZenGround0) [reply gpbft/powertable.go:63]: I find this function confusing enough given that it returns true if p[i] is greater than p[j] and it's a `Less` function

**Review by ZenGround0 — APPROVED** [review]

File: big/int.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Stebalien) [reply big/int.go:1]: We don't want to but... tbh, I might as well just import it for now.

[the remark this answers is not in the document]
Comment (Stebalien) [reply big/int.go]: Done.

File: gpbft/powertable.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Stebalien) [reply gpbft/powertable.go:63]: Eh... I'm just going to leave this for now as it's existing code.

[the remark this answers is not in the document]
Comment (Stebalien) [reply gpbft/powertable.go:63]: Maybe later.

**Review by rvagg — APPROVED** [review]
looks ok to me other than this likely not being a hard protocol breaking change - both sides of this (old and new) will probably be able to decode their own `big.Int` from what they receive, it'll just be wrong, because of that prefix sign byte.

**Comment by Stebalien** [comment]
(discussed on slack)

We revved the protocol version, so anyone running lotus 1.28 won't join the network.

**Pull request merged**